component/app.jsonnet

```
@@ -2,15 +2,29 @@ local kap = import 'lib/kapitan.libjsonnet';
 local inv = kap.inventory();
 local params = inv.parameters.crossplane;
 local argocd = import 'lib/argocd.libjsonnet';
+local on_openshift4 = inv.parameters.facts.distribution == 'openshift4';
+
+local ignore_diff_sa = {
+  group: '',
+  kind: 'ServiceAccount',
+  name: '',
+  jsonPointers: [ '/imagePullSecrets' ],
+};
+
+local ignore_diff_cr = {
+  group: 'rbac.authorization.k8s.io',
+  kind: 'ClusterRole',
+  name: 'crossplane',
+  jsonPointers: [ '/rules' ],
+};
 
 local app = argocd.App('crossplane', params.namespace) {
   spec+: {
-    ignoreDifferences: [ {
-      group: 'rbac.authorization.k8s.io',
-      kind: 'ClusterRole',
-      name: name,
-      jsonPointers: [ '/rules' ],
-    } for name in [ 'crossplane' ] ],
+    ignoreDifferences:
+      if on_openshift4 then
+        [ ignore_diff_sa, ignore_diff_cr ]
+      else
+        [ ignore_diff_cr ],
```

Purely stylistic, you could also do the following:

```suggestion
    ignoreDifferences:
      [ ignore_diff_cr ] +
      if on_openshift4 then
        [ ignore_diff_sa ]
      else
        [ ],
```